core/src/test/java/kafka/test/ClusterInstance.java

```diff
@@ -183,6 +193,67 @@ default Set<GroupProtocol> supportedGroupProtocols() {
 
     //---------------------------[wait]---------------------------//
 
+    @SuppressWarnings("deprecation")
+    default void verifyTopicDeletion(String topic, int partitions) throws InterruptedException {
+        if (!isKRaftTest()) {
```

Comment:
This should be moved into zk instance, since it is not a part of default implementation. FOr example:

```java
        @Override
        public void verifyTopicDeletion(String topic, int partitions) throws InterruptedException {
            org.apache.kafka.test.TestUtils.waitForCondition(
                    () -> !getUnderlying().zkClient().isTopicMarkedForDeletion(topic),
                    String.format("Admin path /admin/delete_topics/%s path not deleted even after a replica is restarted", topic)
            );

            org.apache.kafka.test.TestUtils.waitForCondition(
                    () -> !getUnderlying().zkClient().topicExists(topic),
                    String.format("Topic path /brokers/topics/%s not deleted after /admin/delete_topics/%s path is deleted", topic, topic)
            );
            ClusterInstance.super.verifyTopicDeletion(topic, partitions);
        }
```